projects/dev/src/app/group/group-content/group-content.component.ts

```diff
@@ -1,39 +1,66 @@
-import { Component, OnInit, ElementRef, EventEmitter, Input, Output } from '@angular/core';
+import { Component, OnInit, Input } from '@angular/core';
 import * as _ from 'lodash';
+import { Group } from '../../shared/models/group.model';
+import { GroupSettingsComponent } from './group-settings/group-settings.component';
+import { GroupAdministrationComponent } from './group-administration/group-administration.component';
+import { GroupOverviewComponent } from './group-overview/group-overview.component';
+import { GroupCompositionComponent } from './group-composition/group-composition.component';
+
+interface NavLink {
+  path: string;
+  label: string;
+}
 
 @Component({
   selector: 'app-group-content',
   templateUrl: './group-content.component.html',
   styleUrls: ['./group-content.component.scss'],
 })
 export class GroupContentComponent implements OnInit {
-  @Input() data;
-  @Input() empty;
 
-  @Output() expandWholeWidth = new EventEmitter<void>();
+  @Input() group: Group;
+  activeTab = 0;
+  navLinks: NavLink[] = [
+    {
+      path: './',
+      label: 'Overview'
+    },
+    {
+      path: 'members',
+      label: 'Composition'
+    },
+    {
+      path: 'managers',
+      label: 'Administration'
+    },
+    {
+      path: 'settings',
+      label: 'Settings'
+    }
+  ];
 
-  constructor(private elementRef: ElementRef) {}
+  constructor() { }
 
-  ngOnInit() {}
+  ngOnInit() {
+  }
 
-  onTabChange(e) {
-    const tabs = this.elementRef.nativeElement.querySelectorAll(
-      '.mat-tab-labels .mat-tab-label'
-    );
-    const activeTab = this.elementRef.nativeElement.querySelector(
-      '.mat-tab-labels .mat-tab-label.mat-tab-label-active'
-    );
-    tabs.forEach((tab) => {
-      tab.classList.remove('mat-tab-label-before-active');
-    });
+  canShowTab(idx: number) {
+    return idx === 0 || idx === 1 || this.group.canMangeMembershipAndGroup();
+  }
 
-    const i = _.findIndex(tabs, activeTab);
-    if (i > 0) {
-      tabs[i - 1].classList.add('mat-tab-label-before-active');
-    }
+  canShowTabBar() {
+    return this.canShowTab(this.activeTab);
```

Comment:
To get the current active tab on group content I need to use that `activeTab` variable.
To hide/show tab bar, I need to know what is the active route url is. That's because entering url directly on browser can only be detected by the `router-outlet`'s `activated` output event or `Router`'s url observable.

Reply:
So you could already apply the first part of my remark.

For `activeTab`, you don't use it in html. Where do you use it out of `group-content/group-content.component.ts`?

Reply:
It's only used in `group-content.component.ts` file to get what current active tab will be from the url and determine if this url can show the tab bar or not.